internalsdk/pro/pro.go

```diff
@@ -57,29 +59,37 @@ type ProClient interface {
 // NewClient creates a new instance of ProClient
 func NewClient(baseURL string, opts *webclient.Opts) ProClient {
 	if opts.HttpClient == nil {
-		// The default http.RoundTripper used by the ProClient is ParallelForIdempotent which
-		// attempts to send requests through both chained and direct fronted routes in parallel
-		// for HEAD and GET requests and ChainedThenFronted for all others.
-		opts.HttpClient = NewHTTPClient(proxied.ParallelForIdempotent(), opts.Timeout)
+		opts.HttpClient = &http.Client{
+			// The default http.RoundTripper used by the ProClient is ParallelForIdempotent which
+			// attempts to send requests through both chained and direct fronted routes in parallel
+			// for HEAD and GET requests and ChainedThenFronted for all others.
+			Transport: proxied.ParallelForIdempotent(),
+			Timeout:   30 * time.Second,
+		}
 	}
+
 	if opts.OnBeforeRequest == nil {
 		opts.OnBeforeRequest = func(client *resty.Client, req *http.Request) error {
-			prepareProRequest(req, common.ProAPIHost, opts.UserConfig())
+			prepareProRequest(req, opts.UserConfig())
 			return nil
 		}
 	}
+
+	if opts.BaseURL == "" {
+		opts.BaseURL = baseURL
+	}
```

Comment:
I'm specifying the URL / path prefix here instead of in `prepareProRequest` (where the request URL was previously edited manually).  If I'm understanding `webclient.Opts.BaseURL` correctly, then I think this should work.  I'll confirm with manual testing.